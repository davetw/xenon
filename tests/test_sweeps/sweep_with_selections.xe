# This sweep tests whether sweeps for a selection of objects produce the
# correct values when there is a sweep on the same parameter for a larger
# scope. That is:
#
# `sweep unrolling for ** from 1 to 4 expstep` should be applied to all loops
# except for those under aes_aes.aes256_encrypt_ecb, which have their own sweep
# commands and come after the sweep on **.

begin sweep mysweep

use xenon.tests.machsuite

generate configs

# Set parameters.
set output_dir "test"
set source_dir "test"
set simulator "aladdin"
set memory_type "spad"

set pipelining 1
set partition_type "cyclic"
set partition_type for aes_aes.k "block"
sweep unrolling for ** from 1 to 4 expstep 2
sweep unrolling for aes_aes.aes256_encrypt_ecb.* from 2 to 8 expstep 2
sweep unrolling for aes_aes.aes256_encrypt_ecb.ecb2 from 1 to 3 linstep 1

end sweep


# Same sweep as above, but we've put the more selective sweep command before
# the global one.  The global one should override the more selective one.
begin sweep mysweep2

use xenon.tests.machsuite

generate configs

# Set parameters.
set output_dir "test"
set source_dir "test"
set simulator "aladdin"
set memory_type "spad"

set pipelining 1
set partition_type "cyclic"
set partition_type for aes_aes.k "block"
sweep unrolling for aes_aes.aes256_encrypt_ecb.* from 2 to 8 expstep 2
sweep unrolling for aes_aes.aes256_encrypt_ecb.ecb2 from 1 to 3 linstep 1
sweep unrolling for ** from 1 to 4 expstep 2

end sweep

# Same sweep as above, but now we set a few parameters to constant values.
# after all the sweep commands.
begin sweep mysweep3

use xenon.tests.machsuite

generate configs

# Set parameters.
set output_dir "test"
set source_dir "test"
set simulator "aladdin"
set memory_type "spad"

set pipelining 1
set partition_type "cyclic"
set partition_type for aes_aes.k "block"
sweep unrolling for aes_aes.aes256_encrypt_ecb.* from 2 to 8 expstep 2
sweep unrolling for aes_aes.aes256_encrypt_ecb.ecb2 from 1 to 3 linstep 1
sweep unrolling for ** from 1 to 4 expstep 2
set unrolling for aes_aes.aes256_encrypt_ecb.ecb2 10
set unrolling for aes_aes.aes_subBytes.sub 12


end sweep
