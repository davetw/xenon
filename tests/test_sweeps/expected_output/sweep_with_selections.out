{ 'SweepableView(DesignSweep("mysweep"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                     'partition_type': 'block'},
                                                                                      'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 2},
                                                                                                                                         'SweepableView(Loop("ecb2"))': { 'unrolling': 1},
                                                                                                                                         'SweepableView(Loop("ecb3"))': { 'unrolling': 2}},
                                                                                      'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 1}},
                                                                                      'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 1}},
                                                                                      'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 1}},
                                                                                      'cycle_time': 1,
                                                                                      'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                     'partition_type': 'block'},
                                                                                      'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 4},
                                                                                                                                         'SweepableView(Loop("ecb2"))': { 'unrolling': 2},
                                                                                                                                         'SweepableView(Loop("ecb3"))': { 'unrolling': 4}},
                                                                                      'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 2}},
                                                                                      'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 2}},
                                                                                      'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 2}},
                                                                                      'cycle_time': 1,
                                                                                      'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                       'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                     'partition_type': 'block'},
                                                                                      'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                      'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 8},
                                                                                                                                         'SweepableView(Loop("ecb2"))': { 'unrolling': 3},
                                                                                                                                         'SweepableView(Loop("ecb3"))': { 'unrolling': 8}},
                                                                                      'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 4}},
                                                                                      'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 4}},
                                                                                      'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 4}},
                                                                                      'cycle_time': 1,
                                                                                      'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep2"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 1},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 1},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 1}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep2"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 2},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 2},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 2}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep2"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 4},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 4},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 4}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep3"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 1},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 10},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 1}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 12}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep3"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 2},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 10},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 2}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 12}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
{ 'SweepableView(DesignSweep("mysweep3"))': { 'SweepableView(Benchmark("aes-aes"))': { 'SweepableView(Array("buf"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("ctx"))': { 'partition_factor': 1,
                                                                                                                        'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("k"))': { 'partition_factor': 1,
                                                                                                                      'partition_type': 'block'},
                                                                                       'SweepableView(Array("rcon"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Array("sbox"))': { 'partition_factor': 1,
                                                                                                                         'partition_type': 'cyclic'},
                                                                                       'SweepableView(Function("aes256_encrypt_ecb"))': { 'SweepableView(Loop("ecb1"))': { 'unrolling': 4},
                                                                                                                                          'SweepableView(Loop("ecb2"))': { 'unrolling': 10},
                                                                                                                                          'SweepableView(Loop("ecb3"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_addRoundKey"))': { 'SweepableView(Loop("addkey"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_addRoundKey_cpy"))': { 'SweepableView(Loop("cpkey"))': { 'unrolling': 4}},
                                                                                       'SweepableView(Function("aes_subBytes"))': { 'SweepableView(Loop("sub"))': { 'unrolling': 12}},
                                                                                       'cycle_time': 1,
                                                                                       'pipelining': 1}}}
